### PR TITLE
SONARJAVA-2861: message from S1598 should be explicit when folder name contains dots

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java [hunk 1]
```diff
@@ -27,12 +29,12 @@
 import org.sonar.plugins.java.api.tree.CompilationUnitTree;
 import org.sonar.plugins.java.api.tree.PackageDeclarationTree;
 
-import java.io.File;
-
 @Rule(key = "S1598")
 public class MismatchPackageDirectoryCheck extends BaseTreeVisitor implements JavaFileScanner {
 
   private JavaFileScannerContext context;
+  private static final String MESSAGE = "This file \"${0}\" should be located in \"${1}\" directory, not in \"${2}\"";
```
Comment (alban-auzeill) [inline java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java]: I know that you dream to be a millionaire, but it's not a reason to add *$$$* everywhere.
MessageFormat.format arguments should be `{0}`, `{1}`, `{2}`  and not  `${0}`, `${1}`, `${2}`

File: java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java [hunk 2]
```diff
@@ -49,8 +51,14 @@ public void visitCompilationUnit(CompilationUnitTree tree) {
       File javaFile = context.getFile();
       String dir = javaFile.getParent();
       if (!dir.endsWith(packageName)) {
-        context.reportIssue(
-          this, packageDeclaration.packageName(), "This file \"" + javaFile.getName() + "\" should be located in \"" + packageName + "\" directory, not in \"" + dir + "\".");
+        String dirWithoutDots = dir.replace(".", "/");
```
Comment (alban-auzeill) [inline java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java]: Replace "/" by File.separator

File: java-checks/src/test/files/checks/mismatchPackage/with.dots/PackageWithDots.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package mismatchPackage.with.dots; // Noncompliant {{This file "$PackageWithDots.java" should be located in "$mismatchPackage/with/dots" directory, not in "$src/test/files/checks/mismatchPackage/with.dots"(Do not use dots in directory names).}}
```
Comment (alban-auzeill) [inline java-checks/src/test/files/checks/mismatchPackage/with.dots/PackageWithDots.java]: This test should fail on windows, I don't see a simple solution to test the message. We could probably accept that the message is not tested.

Comment (andrea-guarino-sonarsource) [reply java-checks/src/test/files/checks/mismatchPackage/with.dots/PackageWithDots.java]: Good point, I'll remove the message then

File: java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java (hunk 1 above)
Comment (andrea-guarino-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java]: :sweat_smile: thanks, I mixed two syntaxes

**Pull request merged**